### PR TITLE
elfutils: Add support for debuginfod

### PR DESCRIPTION
**Pull request by hainest**
Requires #18226 be merged before this can be merged.

### CODE REVIEW

**Review by adamjstewart — APPROVED** [review]

File: var/spack/repos/builtin/packages/elfutils/package.py [hunk 1]
```diff
@@ -43,13 +43,25 @@ class Elfutils(AutotoolsPackage, SourcewarePackage):
...
+            description='Enable libdebuginfod support.')
+
     depends_on('bzip2', type='link', when='+bzip2')
     depends_on('xz',    type='link', when='+xz')
     depends_on('zlib',  type='link')
     depends_on('gettext', when='+nls')
     depends_on('m4',    type='build')
 
+    # debuginfod has extra dependencies
+    # NB: Waiting on an elfutils patch before we can use libmicrohttpd@0.9.71
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/elfutils/package.py:57]: Can you clarify this comment? If we merge the PR would `elfutils+debuginfod` be functional now that #18226 has been merged?

**Review by alalazo — CHANGES_REQUESTED** [review]
Requested a clarification on a comment. Code LGTM

File: var/spack/repos/builtin/packages/elfutils/package.py (hunk 1 above)
Comment (hainest) [reply var/spack/repos/builtin/packages/elfutils/package.py:57]: Yes, `elfutils+debuginfod` will be functional, but it cannot build with libmicrohttpd@0.9.71 until an upstream elfutils patch fixes an API change.

**Comment by hainest** [comment]
@alalazo Now that libmicrohttpd has been merged, this is passing all unit tests. It's ready to be merged when you have time.

**Review by alalazo — APPROVED** [review]

**Pull request merged**